File: UXWebExam/ClientApp/src/classes/Car.ts
export class Car {
    public name: string;
    public price: number;
    public type: Type;
    public imageURL: string;
    public street: string;
    public houseNumber: string;
    public postcode: string;
    public city: string;
  
  
    constructor(name: string ,price:number,type:Type,imageURL:string, street:string, houseNumber:string, postcode:string, city:string) {
      this.name = name;
      this.price = price;
      this.type = type;
      this.imageURL = imageURL;
      this.street= street;
      this.houseNumber= houseNumber;
      this.postcode=postcode;
      this.city=city;
    }

 
}

export enum Type {
  Electric = "Electric",
  Petrol = "Petrol",
  Diesel = "Diesel"
}

File: UXWebExam/Data/Car.cs
namespace UXWebExam.Data;

public class Car
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public CarType Type { get; set; }

    public decimal Price { get; set; }

    public string? ImageUrl { get; set; }

    public string Street { get; set; } = null!;

    public string HouseNumber { get; set; } = null!;

    public int PostalCode { get; set; }

    public string City { get; set; } = null!;

    public double Longitude { get; set; }

    public double Latitude { get; set; }

    // Navigation properties

    public ICollection<Booking> Bookings { get; set; } = null!;
}
